service/history/shard/context_impl.go

```diff
@@ -684,6 +686,35 @@ func (s *ContextImpl) AppendHistoryEvents(
 	return size, err0
 }
 
+func (s *ContextImpl) DeleteWorkflowExecution(
+	curRequest *persistence.DeleteCurrentWorkflowExecutionRequest,
```

Comment:
These two requests have the same fields. Should it construct one from the other, or have both as args?

Reply:
can you simply use WorkflowKey from definition package

Reply:
I can, but first let's decide exactly which calls should be moved into context.. question above (we might need the branch token also)

Reply:
We should have only one arg, and use WorkflowKey.

Reply:
If we delete history also we need a branch token.

Reply:
I thought we didn't need a branch token if we were specifying deletion of ALL the history for a Workflow?

Reply:
As far as I can tell, the branch token is passed down pretty deeply into persistence. If there's a way to delete history without it, it's not obvious to me. I don't see any API to do it, at least.